activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields.any?
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
+            Passing arguments to the select method when used with a block is deprecated and will be removed in future version of Rails.
```

Comment:
I think it should really be a specific version.

But the overall wording could still be improved, IMO... "Passing arguments to the select method when used with a block" *is* what's happening, but (ignoring the fact it feels run-on) it under delivers on helping the caller know what's actually going on.

A slightly-sleepy attempt: "When called with a block, select ignores other arguments. This behavior is deprecated, and it will raise an ArgumentError in Rails 5.1."

That is: calling out the fact the arguments are ignored (so they know it's safe to just delete them from their call), and then looking for an alternative to "removed" because it feels confusing given there's no real functionality to take away.